sonic_platform_base/component_base.py

```diff
@@ -50,3 +78,30 @@ def install_firmware(self, image_path):
             A boolean, True if install was successful, False if not
         """
         raise NotImplementedError
+
+    def update_firmware(self, image_path):
+        """
+        Updates firmware of the component.
+        It's API's responsibility to complete firmware update
+        in case some extra steps are required (e.g., reboot, power cycle, etc.)
```

Comment:
Can you add more detail to this docstring and/or the docstring of `install_firmware` above to distinguish the differences in functionality between these two functions? Currently they appear as though they accomplish the same thing.

Reply:
@jleveque The difference between these two is that originally `install_firmware` was implemented to do installation only (we have return value True/False). The update process is a bit different - it assumes not only firmware installation but also all the rest steps which are needed to load the new version. In this case there is no chance to handle the return code, since there might be power cycle or cold reboot as part of update operation. The only reliable way to handle the errors is to throw the exceptions. For this purpose we need a dedicated API - `update_firmware`. A call to this new API will guarantee that firmware update has been completed and no user interaction is required.

The `install_firmware` is still needed to provide user full control on firmware installation process in case delayed update is expected: for example if the power cycle/cold reboot should be done later.

Reply:
@nazariig: Can you update the docstrings of these functions to add some of that detail so whomever is implementing them understands what each does? Currently, the docstrings are very similar; it's not quite clear what the difference is.

Reply:
@jleveque sure. I will extend the docstrings and add more info.

Reply:
@jleveque done. Please review the latest changes.

Reply:
Much better. Thanks!